apis/python/src/tiledb/vector_search/ingestion.py

```diff
@@ -1632,10 +1632,30 @@ def ingest_vamana(
 
         ctx = vspy.Ctx(config)
         index = vspy.IndexVamana(ctx, index_group_uri)
-        data = vspy.FeatureVectorArray(ctx, parts_array_uri, ids_array_uri)
+        data = vspy.FeatureVectorArray(
+            ctx,
+            parts_array_uri,
+            ids_array_uri,
+            0,
+            index_timestamp if index_timestamp else 0,
+        )
         index.train(data)
         index.add(data)
-        index.write_index(ctx, index_group_uri, index_timestamp)
+        # If we just created the index, we need to overwrite the index metadata. This is because if
+        # we are creating a new index during ingestion, we will call write_index() twice. In this
+        # case we should overwrite the previous ingestion_timestamps and other lists, otherwise we will
+        # end up with ingestion_timestamps = [0, timestamp] and base_sizes = [0, initial size],
+        # whereas other Python indexes will end up with ingestion_timestamps = [timestamp] and
+        # base_sizes = [initial size]. This becomes problematic when we try to modify the index
+        # later (i.e. through update / clear / etc).)
+        overwrite_metadata_list = not arrays_created
```

Comment:
This check is a bit subtle and I am not sure it will work consistently for all cases. 

For example, if we do `vamana_index.create` and then `ingest` this will be `false` even if it is the first ingestion. 

Maybe a more consistent check would be to check `base_size==0 && base_ingestion_timestamp==0`. 

I would also suggest to do such checks in C++ and not pass the values from Python in order for the C++ code to be self-contained.

Reply:
Sure, good suggestion, updated.